### PR TITLE
[Fix] Revise override in init_cfg

### PR DESCRIPTION
**Pull request by MeowZheng**
deep copy `override_` to improve the robustness

### CODE REVIEW

**Review by ZwwWayne — APPROVED** [review]

File: mmcv/cnn/utils/weight_init.py [hunk 1]
```diff
@@ -376,11 +376,12 @@ def _initialize_override(module, override, cfg):
     override = [override] if isinstance(override, dict) else override
 
     for override_ in override:
-        if 'type' not in override_.keys():
-            override_.update(cfg)
-        name = override_.pop('name', None)
+        cp_cfg = copy.deepcopy(override_)
```
Comment (ZwwWayne) [inline mmcv/cnn/utils/weight_init.py:379]: Suggest using unit tests to verify that the original config is not modified.

File: tests/test_cnn/test_weight_init.py [hunk 2]
```diff
@@ -362,3 +401,21 @@ def test_initialize():
...
+    # test override with args except type
+    with pytest.raises(ValueError):
+        init_cfg = dict(
+            type='Constant',
+            val=1,
+            bias=2,
+            override=dict(name='conv2d_2', val=3, bias=4))
+        initialize(foonet, init_cfg)
+
+    # test override wiout name
```
Comment (ZwwWayne) [inline tests/test_cnn/test_weight_init.py:414]: without

File: tests/test_cnn/test_weight_init.py [hunk 3]
```diff
@@ -362,3 +401,21 @@ def test_initialize():
                 dict(type='Constant', name='conv2d_3', val=5, bias=6)
             ])
         initialize(foonet, init_cfg)
+
+    # test override with args except type
```
Comment (ZwwWayne) [inline tests/test_cnn/test_weight_init.py:405]: except type?

Comment (MeowZheng) [reply tests/test_cnn/test_weight_init.py]: except type key

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**